api/src/main/java/org/pmiops/workbench/exfiltration/EgressLogService.java

```diff
@@ -0,0 +1,162 @@
+package org.pmiops.workbench.exfiltration;
+
+import com.google.cloud.bigquery.BigQuery.QueryResultsOption;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.QueryJobConfiguration;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.TableResult;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+import javax.annotation.Nullable;
+import javax.inject.Provider;
+import org.pmiops.workbench.api.BigQueryService;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.model.DbEgressEvent;
+import org.pmiops.workbench.exceptions.ServerErrorException;
+import org.pmiops.workbench.model.AuditEgressRuntimeLogEntry;
+import org.pmiops.workbench.model.AuditEgressRuntimeLogGroup;
+import org.pmiops.workbench.model.SumologicEgressEvent;
+import org.pmiops.workbench.utils.FieldValues;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EgressLogService {
+  /**
+   * Scan logs ahead of the egress window by this amount, in the even that the egress activity
+   * started before or near the beginning. This is not guaranteed to catch all relevant logs.
+   */
+  public static final Duration LOG_START_OFFSET = Duration.ofMinutes(5L);
+
+  public static final int RUNTIME_LOG_LIMIT = 1000;
+
+  public static final List<EgressTerraRuntimeLogPattern> terraRuntimeLogPatterns =
+      ImmutableList.of(
+          new EgressTerraRuntimeLogPattern("Notebook Interactions", "%.ipynb%"),
+          new EgressTerraRuntimeLogPattern("File downloads", "%download%"),
+          new EgressTerraRuntimeLogPattern("Errors", "%error%"));
+
+  @Autowired private Provider<WorkbenchConfig> workbenchConfigProvider;
```

Comment:
nit, in general constructor Injection is preferred than parameter injection.

Reply:
Why? Are you referencing a style guide?

In general, this pattern reduces the boilerplate. Downside is that it prevents plain construction of the object, but in our unit tests directly constructing a service like this is not best practice.

Reply:
I know that is prefered in Guice, not sure if the same reason applies in Spring. (easier to )
go/di-practices/bugpatterns/BadInject#inject-on-fieldmethod-and-constructor
go/di-practices/injection-site#constructor-injection

Reply:
Fair point - I think it does apply here, though the extra boilerplate feels bad, this is just the cost of Java. Switched to this pattern.